### DISCUSSION TITLE
Lesson 7 13 - Foundry test cheatcodes testFundUpdatesFundedDataStructure

### DISCUSSION THREAD

**Discussion by neo-t-maredi** [Cyfrin/foundry-full-course-cu#1708] [Q&A]
Hi All! 

I'm currently on lesson/part 13 of the Foundry Fundamentals course. I'm currently having an error (7576) thrown as below: 
![Screenshot 2024-05-13 092229](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/8ae8a7a9-4df9-4f08-8c30-bdc63a34aa12)
![Screenshot 2024-05-13 092229](https://github.com/Cyfrin/foundry-full-course-f23/assets/146746647/7b3eb319-8369-4e72-a500-d7bb2df7c0e5)

Please find below my code for FundMe.sol 

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
import {PriceConverter} from "./PriceConverter.sol";

error FundMe__NotOwner();

contract FundMe {
    using PriceConverter for uint256;

    mapping(address => uint256) public s_addressToAmountFunded;
    address[] public s_funders;

    // Could we make this constant?  /* hint: no! We should make it immutable! */
    address public /* immutable */ i_owner;
    uint256 public constant MINIMUM_USD = 5e18;
    AggregatorV3Interface private s_priceFeed;
    
    constructor(address priceFeed) {
        i_owner = msg.sender;
        s_priceFeed = AggregatorV3Interface(priceFeed);

    }

    function fund() public payable {
        require(msg.value.getConversionRate(s_priceFeed) >= MINIMUM_USD, "You need to spend more ETH!");
        // require(PriceConverter.getConversionRate(msg.value) >= MINIMUM_USD, "You need to spend more ETH!");
        s_addressToAmountFunded[msg.sender] += msg.value;
        s_funders.push(msg.sender);
    }
    
    function getVersion() public view returns (uint256){
       return s_priceFeed.version();
    }
    
    modifier onlyOwner {
        // require(msg.sender == owner);
        if (msg.sender != i_owner) revert FundMe__NotOwner();
        _;
    }
    
    function withdraw() public onlyOwner {
        for (
            uint256 funderIndex=0; 
            funderIndex < s_funders.length; 
            funderIndex++
        ) {
            address funder = s_funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // // transfer
        // payable(msg.sender).transfer(address(this).balance);
        
        // // send
        // bool sendSuccess = payable(msg.sender).send(address(this).balance);
        // require(sendSuccess, "Send failed");

        // call
        (bool callSuccess, ) = payable(msg.sender).call{value: address(this).balance}("");
        require(callSuccess, "Call failed");
    }
    // Explainer from: https://solidity-by-example.org/fallback/
    // Ether is sent to contract
    //      is msg.data empty?
    //          /   \ 
    //         yes  no
    //         /     \
    //    receive()?  fallback() 
    //     /   \ 
    //   yes   no
    //  /        \
    //receive()  fallback()

    fallback() external payable {
        fund();
    }

    receive() external payable {
        fund();
    }

    /**
     * View / Pure functions (Getters)
     */

     function getAddressToAmountFunded(
        address fundingAddress
    ) external view returns (uint256) {
         return s_addressToAmountFunded[fundingAddress];
     }

     function getFunder() external view returns (address) {
         return s_funders[index];
     }
}

**Comment by neo-t-maredi** [comment]
Not getting errors when writing the code and also enabled github co-pilot but still running into these issues.

**Comment by muhammadjehanzaib** [comment]
function getFunder() external view returns (address) {
     return s_funders[index];
 }

here is the issue. 
your arrgument is missing.
Here is the solution

'''solidity
 function getFunder(uint256 index) external view returns (address) {
     return s_funders[index];
 }
'''

**Reply by EngrPips** [reply]
Hello @neo-t-maredi, This should solve your problem.
